### PR TITLE
OEL-3327: Improve tests coverage in 3.x

### PR DESCRIPTION
**Pull request by AaronGilMartinez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ For Drupal projects, `drupal-conventions.yml` should be imported instead.
 
 OE Components should use `oe-components-conventions.yml`
 
+Projects using Drupal rules based conventions like phpcs in `drupal-conventions` and `oe-components-conventions.yml` have to require Coder via Composer:
+
+```bash
+composer require --dev drupal/coder
+```
+
```
Comment (drishu) [inline README.md:31]: could you also fix
"The library is provides docker-compose" to "The library provides docker-compose"

File: tests/PhpCodeSnifferTestBase.php [hunk 2]
```diff
@@ -69,4 +71,38 @@ protected function getFailures(TaskResultInterface $result)
...
+        // Check the result code and possible errors.
+        $this->assertEquals($expectedResultCode, $result->getResultCode());
+        $this->assertEquals($expectedFailures, $this->getFailures($result->first()));
+    }
+
+    /**
+     * Provides test cases for testing the PHP_CodeSniffer task.
+     *
+     * @return array
+     *      Test data.
```
Comment (drishu) [inline tests/PhpCodeSnifferTestBase.php]: isn't this too many spaces?

File: tests/PhpCodeSnifferComponentTest.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace OpenEuropa\CodeReview\Tests;
```
Comment (drishu) [inline tests/PhpCodeSnifferComponentTest.php]: should we take this chance to fix the namespace, move tests under tests/src folder ?

**Review by drishu — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (AaronGilMartinez) [reply README.md:31]: Fixed

File: tests/PhpCodeSnifferComponentTest.php (hunk 3 above)
Comment (AaronGilMartinez) [reply tests/PhpCodeSnifferComponentTest.php]: If we move tests the we have to fix how fixture files are loaded. They use current file location to get the path.

Removed autoload.

File: tests/PhpCodeSnifferTestBase.php (hunk 2 above)
Comment (AaronGilMartinez) [reply tests/PhpCodeSnifferTestBase.php]: Indeed this seems odd, fixed manually. The component uses the PHP library convention to check code. Right now there is no rule for doc commenting identation in PHP library rules. 
Replaced deprecated PSR-2 with PSR-12.

File: tests/PhpCodeSnifferComponentTest.php (hunk 3 above)
Comment (drishu) [reply tests/PhpCodeSnifferComponentTest.php]: out of scope so its ok, but lets create a ticket for it, maybe some day we will do it

File: tests/fixtures/phpcs/DrupalClassCorrect.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+   * Using \Drupal here but it should be injected instead.
+   */
+  public function test(): bool {
+    return TRUE;
+  }
+
+  /**
+   * Tests method with docblock and PHP attribute.
+   */
+  public function aMethodWithDocblockAndPhpAttribute(): void {
```
Comment (drishu) [inline tests/fixtures/phpcs/DrupalClassCorrect.php:23]: Methods are descriptive, nice, but we need to have the correct naming

File: tests/fixtures/phpcs/DrupalClassCorrect.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Drupal\testmodule;
+
+use Drupal\BlockBase;
+
+/**
+ * Some service.
```
Comment (drishu) [inline tests/fixtures/phpcs/DrupalClassCorrect.php]: If it where a service, it would be named as such

Comment (AaronGilMartinez) [reply tests/fixtures/phpcs/DrupalClassCorrect.php]: Fixed

File: tests/fixtures/phpcs/DrupalClassCorrect.php (hunk 4 above)
Comment (AaronGilMartinez) [reply tests/fixtures/phpcs/DrupalClassCorrect.php:23]: Fixed

File: tests/PhpCodeSnifferComponentTest.php (hunk 3 above)
Comment (AaronGilMartinez) [reply tests/PhpCodeSnifferComponentTest.php]: I've changed my mind on this, we are improving tests, moved them to src, I though was a bigger change.

**Review by drishu — DISMISSED** [review]

File: README.md [hunk 6]
```diff
@@ -115,7 +121,7 @@ Parameters available from base-conventions in new version are the same, no chang
 
 ### Installation
 
-The library is provides docker-compose configuration file to be used for development.
+The library provides docker-compose configuration file to be used for development.
```
Comment (brummbar) [inline README.md]:
```suggestion
The library provides a docker-compose configuration file to be used for development.
```

File: tests/fixtures/phpcs/DrupalClassIncorrect.php [hunk 7]
```diff
@@ -5,9 +5,9 @@
...
 
 /**
- * Some service.
+ * Some Drupal class.
  */
-class DrupalClass extends BlockBase {
+class DrupalClassIncorrect extends BlockBase {
 
   /**
    * Using \Drupal here but it should be injected instead.
```
Comment (brummbar) [inline tests/fixtures/phpcs/DrupalClassIncorrect.php]: Is this true?

File: tests/src/AbstractTest.php [hunk 8]
```diff
@@ -33,16 +31,17 @@ protected function getContainer(string $configuration): Container
     {
         // Create a GrumPHP configuration file to use in the test. Check if a configuration specific file exists, or
         // fall back to a generic template file.
-        $filename = file_exists(__DIR__ . "/config/$configuration.yml.dist")
-        ? __DIR__ . "/config/$configuration.yml.dist"
-        : __DIR__ . '/config/grumphp.yml.dist';
+        $realpath = realpath(__DIR__ . '/..');
+        $filename = file_exists("$realpath/config/$configuration.yml.dist")
+        ? "$realpath/config/$configuration.yml.dist"
+        : "$realpath/config/grumphp.yml.dist";
```
Comment (brummbar) [inline tests/src/AbstractTest.php:37]: This needs to be indented.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (AaronGilMartinez) [reply README.md]: Changed

File: tests/fixtures/phpcs/DrupalClassIncorrect.php (hunk 7 above)
Comment (AaronGilMartinez) [reply tests/fixtures/phpcs/DrupalClassIncorrect.php]: Nope, fixed

File: tests/src/AbstractTest.php (hunk 8 above)
Comment (AaronGilMartinez) [reply tests/src/AbstractTest.php:37]: Curious thing, if ident, phpcs sniffer complains. We are applying a generic rule for identation `Generic.Arrays.ArrayIndent` and it seems it doesn't take multiple lines operator as a one level more. Not sure if this exact rule is provided by another Standard, this is familiar to me but could be part of Drupal rules.

Comment (brummbar) [reply tests/src/AbstractTest.php:37]: Oki

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**